### PR TITLE
Fix error when trying to translate Serbian using the BO interface

### LINKED ISSUE #18062: Cannot translate Serbian in 1.7.6.0
#### Describe the bug

Cannot access the translation interface for the Serbian language:
Screencap 1.7.6

#### Expected behavior

You should be able to access like in < 1.7.5

Screencap 1.7.5

#### Steps to Reproduce

1. Install serbian
2. Go to International > Translations
3. Try to translate BO wordings in Serbian

#### Additional information

* PrestaShop version: 1.7.6.0 and later
* PHP version: Indifferent

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | This change makes it so that it is possible to translate Serbian again using the BO interface. Read below for more.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixed #18062
| How to test?  | See steps in issue.

## More information

In 1.7.6, a [change was introduced](https://github.com/PrestaShop/PrestaShop/pull/11321/files#diff-f0c2048d6122a903ce9515acfd72098dR74-R77) where the locale code provided to the controller in charge of listing the translations to display would be verified against the server's ICU list instead of the list of installed languages. 

This was a double mistake because not only would it let a non-installed locale pass through (it would throw an unhandled exception later in code resulting in a 500 error), but it would also invalidate nonstandard locale codes like `sr-CS` (which we use for Serbian for [some unknown reason](https://github.com/PrestaShop/PrestaShop/issues/17890#issuecomment-596583744)).

This PR changes this verification so that the locale code is compared against the list of installed locales.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18064)

### CODE REVIEW

**Comment by matks** [comment]
Some unit tests fail

**Review by Progi1984 — DISMISSED** [review]

**Comment by eternoendless** [comment]
Tests have been fixed!

**Review by Progi1984 — DISMISSED** [review]

File: src/PrestaShopBundle/Exception/InvalidLanguageException.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace PrestaShopBundle\Exception;
+
+class InvalidLanguageException extends \RuntimeException
+{
+    const LOCALE_NOT_FOUND = 1;
+
+    public static function localeNotFound($locale)
+    {
+        return new self(sprintf('The locale "%s" is not available', $locale), self::LOCALE_NOT_FOUND);
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Exception/InvalidLanguageException.php]:
```suggestion
        return new static(sprintf('The locale "%s" is not available', $locale), self::LOCALE_NOT_FOUND);
```

**Review by PierreRambaud — DISMISSED** [review]
use static instead of self

**Comment by eternoendless** [comment]
Changes applied

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
This PR correctly fix #18062 thanks @eternoendless

**Comment by Progi1984** [comment]
Thanks @eternoendless

**Pull request merged**